charts/bpndiscovery/templates/secret.yaml

```diff
@@ -19,16 +19,33 @@
 
 {{- $deployment_name := printf "cx-%s" .Release.Name }}
 {{- $sec_name := printf "%s-sec" $deployment_name }}
+{{- $secretPassword := randAlphaNum 32 | b64enc }}
 apiVersion: v1
 kind: Secret
 metadata:
   name: {{ $sec_name }}
+  namespace: {{ .Release.Namespace }}
 type: Opaque
+ # use lookup function to check if secret exists
+{{- $secret := (lookup "v1" "Secret" .Release.Namespace .Values.postgresql.auth.existingSecret) }}
 data:
   {{- if .Values.enablePostgres }}
+#  postgres-password: {{ $secretPassword | quote }}
+#  password: {{ $secretPassword | quote }}
   SPRING_DATASOURCE_URL: {{ printf "jdbc:postgresql://%s-postgresql:%v/%s" .Release.Name .Values.postgresql.service.ports.postgresql .Values.postgresql.auth.database | b64enc }}
   SPRING_DATASOURCE_USERNAME: {{ .Values.postgresql.auth.username | b64enc }}
-  SPRING_DATASOURCE_PASSWORD: {{ .Values.postgresql.auth.password | b64enc }}
+  {{ if $secret -}}
+  data:
+    postgres-password: {{ ( .Values.postgresql.auth.password | b64enc )  | default ( index $secret.data "postgres-password" ) | quote }}
+    password: {{ ( .Values.postgresql.auth.password | b64enc )  | default ( index $secret.data "postgres-password" ) | quote }}
+    SPRING_DATASOURCE_PASSWORD: {{ $secretPassword | quote }}
```

Comment:
the secret values needs to be the values from [postgres-password](https://github.com/eclipse-tractusx/sldt-bpn-discovery/pull/71/files#diff-89698cfe1ba306a56daf2d3119401a86d8b361745a2cb49d7ee0ffd7365d442aR39), not just one 'randAlphaNum 32' like [$secretPassword](https://github.com/eclipse-tractusx/sldt-bpn-discovery/pull/71/files#diff-89698cfe1ba306a56daf2d3119401a86d8b361745a2cb49d7ee0ffd7365d442aR22) is currently defined